lib/kapnismology/application_information.rb

```diff
@@ -11,12 +15,26 @@ def trace_id
     def codebase_revision
       latest_commit_info.split(/\s/).first || INFO_UNKNOWN
     rescue Errno::ENOENT, StandardError
-     INFO_UNKNOWN
+      INFO_UNKNOWN
     end
 
     private
+
     def latest_commit_info
-      `git show HEAD --abbrev-commit --oneline`
+      latest_ref_from_git || latest_ref_from_ecs_metadata || nil
+    end
+
+    def latest_ref_from_git
+      result = `#{GIT_SHOW_COMMAND}`
+      $?.success? ? result : nil
+    end
+
+    # This assumes the ImageName features the git ref as the last part of a string delimited by dots.
+    # See spec/support/ecs_metadata.json as well as
+    # https://docs.aws.amazon.com/AmazonECS/latest/developerguide/container-metadata.html
+    def latest_ref_from_ecs_metadata
+      return unless !ECS_CONTAINER_METADATA_FILE&.strip.empty? && File.readable?(ECS_CONTAINER_METADATA_FILE)
+      JSON.parse(File.read(ECS_CONTAINER_METADATA_FILE))["ImageName"].split(".").last[0...7]
```

Comment:
low chances this will change soon but could we catch json parsing exceptions and return nil?

Reply:
https://github.com/JordiPolo/kapnismology/pull/14/commits/daf149a684ff0f543b66baea17cb2e153130ee73